File: MošPosudit.Services/DataBase/Data/Rental.cs
using MošPosudit.Model.Enums;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace MošPosudit.Services.DataBase.Data
{
    public class Rental
    {
        [Key]
        public int Id { get; set; }

        [Required]
        public int UserId { get; set; }

        [Required]
        public DateTime StartDate { get; set; }

        [Required]
        public DateTime EndDate { get; set; }

        [Required]
        public int StatusId { get; set; }

        [Required]
        [Range(0.01, double.MaxValue)]
        public decimal TotalPrice { get; set; }

        public DateTime CreatedAt { get; set; }

        [MaxLength(500)]
        public string? Notes { get; set; }

        [Required]
        public int ToolId { get; set; }
        [ForeignKey("ToolId")]
        public Tool Tool { get; set; }

        public bool IsReturned { get; set; }
        public DateTime? ReturnDate { get; set; }
        public string? ReturnNotes { get; set; }

        public decimal TotalAmount { get; set; }
        public DateTime? UpdatedAt { get; set; }

        // Navigation properties
        [ForeignKey("UserId")]
        public User User { get; set; }
        [ForeignKey("StatusId")]
        public RentalStatus Status { get; set; }
        public ICollection<RentalItem> RentalItems { get; set; }
        public ICollection<PaymentTransaction> Payments { get; set; }
    }
}

File: MosPosudit.UI/shared/lib/models/rental.dart
class RentalModel {
  final int id;
  final int userId;
  final DateTime startDate;
  final DateTime endDate;
  final int statusId;
  final num totalPrice;
  final DateTime createdAt;
  final String? notes;
  final int toolId;
  final bool isReturned;
  final DateTime? returnDate;
  final String? returnNotes;
  final num totalAmount;
  final DateTime? updatedAt;

  RentalModel({
    required this.id,
    required this.userId,
    required this.startDate,
    required this.endDate,
    required this.statusId,
    required this.totalPrice,
    required this.createdAt,
    required this.toolId,
    required this.isReturned,
    required this.totalAmount,
    this.notes,
    this.returnDate,
    this.returnNotes,
    this.updatedAt,
  });

  factory RentalModel.fromJson(Map<String, dynamic> json) => RentalModel(
        id: json['id'],
        userId: json['userId'],
        startDate: DateTime.parse(json['startDate']),
        endDate: DateTime.parse(json['endDate']),
        statusId: json['statusId'],
        totalPrice: json['totalPrice'],
        createdAt: DateTime.parse(json['createdAt']),
        notes: json['notes'],
        toolId: json['toolId'],
        isReturned: json['isReturned'] ?? false,
        returnDate: json['returnDate'] != null ? DateTime.parse(json['returnDate']) : null,
        returnNotes: json['returnNotes'],
        totalAmount: json['totalAmount'] ?? 0,
        updatedAt: json['updatedAt'] != null ? DateTime.parse(json['updatedAt']) : null,
      );

  Map<String, dynamic> toJson() => {
        'id': id,
        'userId': userId,
        'startDate': startDate.toIso8601String(),
        'endDate': endDate.toIso8601String(),
        'statusId': statusId,
        'totalPrice': totalPrice,
        'createdAt': createdAt.toIso8601String(),
        'notes': notes,
        'toolId': toolId,
        'isReturned': isReturned,
        'returnDate': returnDate?.toIso8601String(),
        'returnNotes': returnNotes,
        'totalAmount': totalAmount,
        'updatedAt': updatedAt?.toIso8601String(),
      };
}

